### PR TITLE
feat : added wishlist

### PR DESCRIPTION
**Pull request by Digambar1411**
added wishlist functionality
remove from wishlist
move to wishlist
delete from wishlist

files to check
context/wishlist-contex
pages/wishlist.js
pages/wishlist.css
component/wishlistCard

### CODE REVIEW

File: src/frontend/components/horizontalCard/horizontal-card.js [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 
 const HorizontalCard = (product)=>{
   
-    const {removeFromCart}= useCart()
+    const {removeFromCart , moveToWishlist}= useCart()
+    const { addToWishList } = useWishlist();
+    
+    
     const incrementQuantity=(product)=>{
         // {cart.map((item)=>item._id===product._id) ? {...item, quantity:product.quantity+1} :item
```
Comment (ravikumar0403) [inline src/frontend/components/horizontalCard/horizontal-card.js:14]: remove commented code if not required

File: src/frontend/components/navbar/Navbar.jsx [hunk 2]
```diff
@@ -21,8 +25,9 @@ export function Navbar() {
          
           <div class="icon-w-badge">
```
Comment (ravikumar0403) [inline src/frontend/components/navbar/Navbar.jsx]: change `class` to `className`

File: src/frontend/contexts/wishlist-context.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        setWishlist(newData);
+
+    }
+
+    
+    return ( < WishlistContext.Provider value={{wishlist, addToWishList, removeFromWishList}} > 
+            {children} 
+        </ WishlistContext.Provider >
+    )
+
```
Comment (ravikumar0403) [inline src/frontend/contexts/wishlist-context.js]: remove unnecessary new lines

**Review by ravikumar0403 — APPROVED** [review]

File: src/frontend/components/navbar/Navbar.jsx [hunk 4]
```diff
@@ -21,8 +25,9 @@ export function Navbar() {
          
           <div class="icon-w-badge">        
           <span className="material-icons-outlined md-36" onClick= {()=>navigate("/wishlist")}>favorite_border</span>
-            {/* {wishlist.length>0 && <span class="badge small-badge badge-on-icon">{wishlist.length}</span> } */}
+            {wishlist.length>0 && <span class="badge small-badge badge-on-icon">{wishlist.length}</span> }
```
Comment (amarphule) [inline src/frontend/components/navbar/Navbar.jsx]: use `className` instead `class`

File: src/frontend/components/navbar/Navbar.jsx [hunk 5]
```diff
@@ -21,8 +25,9 @@ export function Navbar() {
...
           </div>
+
           <div class="icon-w-badge">        
             <span className="material-icons md-36" onClick= {()=>navigate("/cart")}>shopping_cart</span>
             {cart.length>0 && <span class="badge small-badge badge-on-icon">{cart.length}</span> }
```
Comment (amarphule) [inline src/frontend/components/navbar/Navbar.jsx]: use `className` instead class

File: src/frontend/components/wishlistCard/wishlistCard.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                    </div>
+                    <span className="rating-span">{product.rating}⭐</span> 
+                </div>
+            </div>
+            <div className="div-padding wishlist-card-btn-div">
+             
+                    <button  className="solid primary-solid" onClick={()=>addToCart(product)}> Add to cart</button> 
+
+                    <div>
+                        <button class="floating-btn delete"><span class="material-icons" onClick={()=>removeFromWishList(product._id)}>delete</span></button>
```
Comment (amarphule) [inline src/frontend/components/wishlistCard/wishlistCard.js]: same as above for `class`

**Review by amarphule — APPROVED** [review]
Nice work, all functionality working

File: src/frontend/components/horizontalCard/horizontal-card.js (hunk 1 above)
Comment (Digambar1411) [reply src/frontend/components/horizontalCard/horizontal-card.js:14]: yeah that one is required will work on that

File: src/frontend/contexts/wishlist-context.js (hunk 3 above)
Comment (Digambar1411) [reply src/frontend/contexts/wishlist-context.js]: thanks done

**Review by manojncsoni007 — COMMENTED** [review]
Nice Work, Everything looks good

**Pull request merged**